CPAC/nuisance/tests/test_bandpass.py

```diff
@@ -46,3 +48,30 @@ def test_read_1D(start_line: int, tmp_path: Path) -> None:
     assert data.shape == (10, 29)
     # all header lines should be captured
     assert len(header) == 5 - start_line
+
+
+@pytest.mark.parametrize(
+    "lowcut, highcut, in_freq, out_freq",
+    [
+        (0.005, 0.05, 0.01, 0.2),
+        (0.01, 0.1, 0.02, 0.15),
+        (0.02, 0.08, 0.04, 0.12),
+    ],
+)
+def test_ideal_bandpass_with_various_cutoffs(lowcut, highcut, in_freq, out_freq):
+    """Test the ideal bandpass filter with various cutoff frequencies."""
+    sample_period = 1.0
```

Comment:
If we parametrize `sample_period`,
```suggestion
@pytest.mark.parametrize("sample_period", [1.0, 1000.0])
@pytest.mark.parametrize(
    "lowcut, highcut, in_freq, out_freq",
    [
        (0.005, 0.05, 0.01, 0.2),
        (0.01, 0.1, 0.02, 0.15),
        (0.02, 0.08, 0.04, 0.12),
    ],
)
def test_ideal_bandpass_with_various_cutoffs(lowcut, highcut, in_freq, out_freq, sample_period):
    """Test the ideal bandpass filter with various cutoff frequencies."""
```

we can also cover [this missed section](https://github.com/FCP-INDI/C-PAC/pull/2237/files#annotation_35639556937)

Reply:
@shnizzedy , I just realized after the failing tests, that the `lowcut`, `highcut` ..etc parameters would be different for each case of `sample_period` so the test failed. 
However, I think the main goal is to cover the `sample_period`, so I have added a test to fix that.